src/drivers/dw/dma.c

```diff
@@ -911,9 +911,6 @@ static int dw_dma_setup(struct dma *dma)
 		return -EIO;
 	}
 
-	for (i = 0; i < dma->plat_data.channels; i++)
-		dma_reg_read(dma, DW_DMA_CHAN_EN);
-
```

Comment:
This was added in eea19e65784adba006d3736d1a79526144f3bf49 by @keyonjie 

@keyonjie any reason why this code was added?

What I find surprising is the code on line 905, something does hint at the need to read after a write, so maybe this was a means to force the state to be consistent?

Reply:
@plbossart my commit eea19e6 is only refining it to support interrupt per channel, the dummy read to CHAN_EN was already there before the commit, see here: https://github.com/thesofproject/sof/commit/eea19e65784adba006d3736d1a79526144f3bf49#diff-e3f71885d1ffdf1924152af66b356ff6L937

I can't recall this clearly, iirc, a dummy read to CHAN_EN is recommended from some reference code. @lgirdwood comment on this?

Reply:
FWIW CI seems to be happy

Reply:
The code line 905 is still somewhat unclear to me. in what case can a DMA setup fail?

Reply:
This may force DMAC to writeback LL descriptors ?